test/Microsoft.DotNet.ProjectModel.Tests/GivenThatIWantToLoadAProjectJsonFile.cs

```diff
@@ -0,0 +1,866 @@
+        private readonly string[] _noWarnings = new[] {"warn1", "warn2"};
+        private readonly string[] _someAdditionalArguments = new[] {"additional argument 1", "additional argument 2"};
+        private readonly VersionRange _versionRange = VersionRange.Parse(Version);
+        private readonly JObject _jsonCompilationOptions;
+        private CommonCompilerOptions commonCompilerOptions;
+
+        public GivenThatIWantToLoadAProjectJsonFile()
+        {
+            var json = new JObject();
+            _emptyProject = GetProject(json);
+            _jsonCompilationOptions = new JObject();
+
+            _jsonCompilationOptions.Add("define", new JArray(_someDefines));
+            _jsonCompilationOptions.Add("nowarn", new JArray(_noWarnings));
+            _jsonCompilationOptions.Add("additionalArguments", new JArray(_someAdditionalArguments));
+            _jsonCompilationOptions.Add("languageVersion", SomeLanguageVersion);
+            _jsonCompilationOptions.Add("outputName", SomeOutputName);
+            _jsonCompilationOptions.Add("platform", SomePlatform);
+            _jsonCompilationOptions.Add("keyFile", SomeKeyFile);
+            _jsonCompilationOptions.Add("debugType", SomeDebugType);
+            _jsonCompilationOptions.Add("allowUnsafe", true);
+            _jsonCompilationOptions.Add("warningsAsErrors", true);
+            _jsonCompilationOptions.Add("optimize", true);
+            _jsonCompilationOptions.Add("delaySign", true);
+            _jsonCompilationOptions.Add("publicSign", true);
+            _jsonCompilationOptions.Add("emitEntryPoint", true);
+            _jsonCompilationOptions.Add("xmlDoc", true);
+            _jsonCompilationOptions.Add("preserveCompilationContext", true);
+
+            commonCompilerOptions = new CommonCompilerOptions
+            {
+                Defines = _someDefines,
+                SuppressWarnings = _noWarnings,
+                AdditionalArguments = _someAdditionalArguments,
+                LanguageVersion = SomeLanguageVersion,
+                OutputName = SomeOutputName,
+                Platform = SomePlatform,
+                KeyFile = SomeKeyFile,
+                DebugType = SomeDebugType,
+                AllowUnsafe = true,
+                WarningsAsErrors = true,
+                Optimize = true,
+                DelaySign = true,
+                PublicSign = true,
+                EmitEntryPoint = true,
+                GenerateXmlDocumentation = true,
+                PreserveCompilationContext = true
+            };
+        }
+
+        [Fact]
+        public void It_does_not_throw_when_the_project_json_is_empty()
+        {
+            var json = new JObject();
+            Action action = () => GetProject(json);
+
+            action.ShouldNotThrow<Exception>();
+        }
+
+        [Fact]
+        public void It_sets_Name_to_the_passed_ProjectName_if_one_is_not_set_in_the_ProjectJson()
+        {
+            _emptyProject.Name.Should().Be(ProjectName);
```

Comment:
can the test better express the intent of this behavior? My expectation was that the ProjectName came from the project file's parent directory name... Instead it is being passed in externally?